src/Development/IDE/Core/Rules.hs

```diff
@@ -679,15 +685,30 @@ isHiFileStableRule = define $ \IsHiFileStable f -> do
     return ([], Just sourceModified)
 
 getModSummaryRule :: Rules ()
-getModSummaryRule = defineEarlyCutoff $ \GetModSummary f -> do
-    dflags <- hsc_dflags . hscEnv <$> use_ GhcSession f
-    (_, mFileContent) <- getFileContents f
-    modS <- liftIO $ evalWithDynFlags dflags $ runExceptT $
-        getModSummaryFromImports (fromNormalizedFilePath f) (textToStringBuffer <$> mFileContent)
-    case modS of
-        Right ms -> do
-            return ( Just (computeFingerprint f dflags ms), ([], Just ms))
-        Left diags -> return (Nothing, (diags, Nothing))
+getModSummaryRule = do
+    defineEarlyCutoff $ \GetModSummary f -> do
+        dflags <- hsc_dflags . hscEnv <$> use_ GhcSession f
+        (fv, mFileContent) <- getFileContents f
+        modTime <- maybe (liftIO getCurrentTime) return $ modificationTime fv
```

Comment:
I have moved the logic to `getFileContents` since all the call sites want it.`GetModIfaceFromDisk` is the only use site that really requires timestamps for the interactive evaluation to work